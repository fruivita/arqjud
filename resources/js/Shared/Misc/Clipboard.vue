<!--
    Componente para cópia de valores para o clipboard.

    Útil para que o usúario possa clicar no botão do componente e seu conteúdo
    ser transferido para a memória, ou seja, um ctrl + c automatizado.

    @link https://vuejs.org/guide/introduction.html
    @link https://tailwindcss.com/docs
    @link https://inertiajs.com/
 -->

<script setup>
import Icone from '@/Shared/Icones/Icone.vue';
import { useClipboard } from '@vueuse/core';

const props = defineProps({
    copiavel: { type: String },
});

const { copy, copied, isSupported } = useClipboard({
    source: props.copiavel,
});
</script>

<template>
    <div class="flex items-center space-y-1">
        <span>{{ copiavel }}</span>

        <button
            v-if="isSupported"
            :disabled="copied"
            @click="copy()"
            class="rounded p-1 opacity-50 ring-primaria-500 transition duration-150 hover:opacity-100 dark:ring-secundaria-500"
        >
            <Icone :nome="copied ? 'check-circle' : 'clipboard'" class="h-4 w-4" />
        </button>
    </div>
</template>
