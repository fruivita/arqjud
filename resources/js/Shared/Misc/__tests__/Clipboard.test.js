/**
 * Testes para o componente Clipboard.
 *
 * @link https://vuejs.org/guide/scaling-up/testing.html
 * @link https://test-utils.vuejs.org/guide/
 * @link https://vitest.dev/
 */

import Clipboard from '@/Shared/Misc/Clipboard.vue';
import { createTestingPinia } from '@pinia/testing';
import { mount } from '@vue/test-utils';
import { beforeEach, describe, expect, test, vi } from 'vitest';

let mountFunction;

let clipboardContents = '';

beforeEach(() => {
    createTestingPinia();

    Object.assign(navigator, {
        clipboard: {
            writeText: vi.fn((text) => {
                clipboardContents = text;
            }),
            readText: vi.fn(() => clipboardContents),
        },
    });

    mountFunction = (options = {}) => {
        return mount(Clipboard, { ...options });
    };
});

describe('Clipboard', () => {
    test('propriedades do componente estão definidas', () => {
        expect(Clipboard.props).toMatchObject({
            copiavel: { type: String },
        });
    });

    test('renderiza o componente default respeitando o snapshot', () => {
        expect(
            mountFunction({
                props: { copiavel: 'foo' },
            }).html()
        ).toMatchSnapshot();
    });

    test('copia o conteúdo para o Clipboard ao clicar no botão', async () => {
        const wrapper = mountFunction({
            props: { copiavel: 'foo' },
        });

        expect(clipboardContents).toBe('');

        await wrapper.get('button').trigger('click');

        expect(clipboardContents).toBe('foo');

        clipboardContents = '';
    });
});
