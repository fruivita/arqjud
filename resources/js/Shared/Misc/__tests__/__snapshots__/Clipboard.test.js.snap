// Vitest Snapshot v1

exports[`Clipboard > renderiza o componente default respeitando o snapshot 1`] = `
"<div class=\\"inline-block\\">
  <div class=\\"flex\\" dusk=\\"toggle\\"><button class=\\"rounded p-1 opacity-50 ring-primaria-500 transition duration-150 hover:opacity-100 dark:ring-secundaria-500\\"><svg class=\\"h-4 w-4\\" fill=\\"currentColor\\" viewBox=\\"0 0 16 16\\" xmlns=\\"http://www.w3.org/2000/svg\\">
        <path d=\\"M4 1.5H3a2 2 0 0 0-2 2V14a2 2 0 0 0 2 2h10a2 2 0 0 0 2-2V3.5a2 2 0 0 0-2-2h-1v1h1a1 1 0 0 1 1 1V14a1 1 0 0 1-1 1H3a1 1 0 0 1-1-1V3.5a1 1 0 0 1 1-1h1v-1z\\"></path>
        <path d=\\"M9.5 1a.5.5 0 0 1 .5.5v1a.5.5 0 0 1-.5.5h-3a.5.5 0 0 1-.5-.5v-1a.5.5 0 0 1 .5-.5h3zm-3-1A1.5 1.5 0 0 0 5 1.5v1A1.5 1.5 0 0 0 6.5 4h3A1.5 1.5 0 0 0 11 2.5v-1A1.5 1.5 0 0 0 9.5 0h-3z\\"></path>
      </svg></button></div>
  <div class=\\"absolute top-0 left-0 z-10 cursor-default rounded-md bg-primaria-600 px-3 py-1.5 text-sm font-bold text-primaria-50 dark:bg-secundaria-600 dark:text-secundaria-50\\" dusk=\\"hint\\" style=\\"display: none;\\"><span></span>
    <div class=\\"absolute h-[8px] w-[8px] rotate-45 bg-primaria-600 dark:bg-secundaria-600\\"></div>
  </div>
</div>"
`;
